src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsaccreditedprogrammesapi/RepositoryTest.kt

```diff
@@ -2,27 +2,22 @@ package uk.gov.justice.digital.hmpps.hmppsaccreditedprogrammesapi
 
 import org.junit.jupiter.api.BeforeEach
 import org.springframework.boot.autoconfigure.domain.EntityScan
-import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest
-import org.springframework.context.annotation.ComponentScan
+import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase
+import org.springframework.boot.test.context.SpringBootTest
 import org.springframework.data.jpa.repository.config.EnableJpaRepositories
 import org.springframework.jdbc.core.JdbcTemplate
 import org.springframework.test.context.ActiveProfiles
-import org.springframework.test.context.ContextConfiguration
 import org.springframework.test.context.transaction.TestTransaction
 import org.springframework.test.jdbc.JdbcTestUtils
+import org.springframework.transaction.annotation.Transactional
 
-private const val COURSE_JPA_REPO_PACKAGE = "uk.gov.justice.digital.hmpps.hmppsaccreditedprogrammesapi.course.jparepo"
-private const val COURSE_DOMAIN_PACKAGE = "uk.gov.justice.digital.hmpps.hmppsaccreditedprogrammesapi.course.domain"
-private const val REFERRAL_JPA_REPO_PACKAGE = "uk.gov.justice.digital.hmpps.hmppsaccreditedprogrammesapi.referral.jparepo"
-private const val REFERRAL_DOMAIN_PACKAGE = "uk.gov.justice.digital.hmpps.hmppsaccreditedprogrammesapi.referral.domain"
-private const val PARTICIPATION_HISTORY_DOMAIN = "uk.gov.justice.digital.hmpps.hmppsaccreditedprogrammesapi.participationhistory.domain"
-private const val PARTICIPATION_HISTORY_JPA_REPO_PACKAGE = "uk.gov.justice.digital.hmpps.hmppsaccreditedprogrammesapi.participationhistory.jparepo"
+private const val BASE_PACKAGE = "uk.gov.justice.digital.hmpps.hmppsaccreditedprogrammesapi"
 
-@DataJpaTest
-@ContextConfiguration(classes = [RepositoryTest::class])
-@ComponentScan(basePackages = [COURSE_JPA_REPO_PACKAGE, REFERRAL_JPA_REPO_PACKAGE, PARTICIPATION_HISTORY_JPA_REPO_PACKAGE])
-@EnableJpaRepositories(basePackages = [COURSE_JPA_REPO_PACKAGE, REFERRAL_JPA_REPO_PACKAGE, PARTICIPATION_HISTORY_JPA_REPO_PACKAGE])
-@EntityScan(basePackages = [COURSE_DOMAIN_PACKAGE, REFERRAL_DOMAIN_PACKAGE, PARTICIPATION_HISTORY_DOMAIN])
+@SpringBootTest
+@Transactional
+@AutoConfigureTestDatabase
+@EnableJpaRepositories(basePackages = [BASE_PACKAGE])
```

Comment:
More of a question than an assertion:
`Application.kt` is annotated with `@SpringBootApplication` and is already in the `uk.gov.justice.digital.hmpps.hmppsaccreditedprogrammesapi` package. Does that make specifying base packages redundant here? I'm not sure if Spring treats it differently because it's a generated class.